internal/factsengine/gatherers/sbddump_test.go

```diff
@@ -0,0 +1,139 @@
+package gatherers
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	mocks "github.com/trento-project/agent/internal/factsengine/gatherers/mocks"
+)
+
+type SBDDumpTestSuite struct {
+	suite.Suite
+}
+
+func TestSBDDumpTestSuite(t *testing.T) {
+	suite.Run(t, new(SBDDumpTestSuite))
+}
+
+func (suite *SBDDumpTestSuite) TestSBDDumpGathererMissingFact() {
+	mockExecutor := new(mocks.CommandExecutor)
+
+	mockOutputFile, _ := os.Open("../../../test/fixtures/gatherers/sbddump.output")
+	mockOutput, _ := ioutil.ReadAll(mockOutputFile)
+	mockExecutor.On("Exec", "sbd", "dump", "-d", "/dev/sdj", "dump").Return(mockOutput, nil)
+
+	c := &SBDDumpGatherer{
+		executor: mockExecutor,
+	}
+
+	factRequests := []FactRequest{
+		{
+			Name:     "nonexistant_timeout",
+			Gatherer: "sbd_dump",
+			Argument: "/dev/sdj:Timeout (nonexistant)",
+		},
+	}
+
+	factResults, err := c.Gather(factRequests)
+
+	expectedResults := []Fact{}
+
+	suite.NoError(err)
+	suite.ElementsMatch(expectedResults, factResults)
+}
+
+func (suite *SBDDumpTestSuite) TestSBDDumpGatherer() {
+	mockExecutor := new(mocks.CommandExecutor)
+
+	mockOutputFile, _ := os.Open("../../../test/fixtures/gatherers/sbddump.output")
+	mockOutput, _ := ioutil.ReadAll(mockOutputFile)
+	mockExecutor.On("Exec", "sbd", "dump", "-d", "/dev/sdj", "dump").Return(mockOutput, nil)
+
+	c := &SBDDumpGatherer{
+		executor: mockExecutor,
+	}
+
+	factRequests := []FactRequest{
+		{
+			Name:     "header_version",
+			Gatherer: "sbd_dump",
+			Argument: "/dev/sdj:Header version",
+		},
+		{
+			Name:     "sector_size",
+			Gatherer: "sbd_dump",
+			Argument: "/dev/sdj:Sector size",
+		},
+		{
+			Name:     "watchdog_timeout",
+			Gatherer: "sbd_dump",
```

Comment:
How would the checks involving `sbd_dump` (1.3.5 and 1.3.6) be represented with the DSL?
Let's say [1.3.5](https://github.com/trento-project/runner/blob/main/runner/ansible/roles/checks/1.3.5/tasks/main.yml) SBD watchdog timeout is set to `expectedValue` (60)

Translating the current implementation to 
> all the devices defined in `SBD_DEVICE` must have the same expected watchdog timeout.
>
it seems that we have to deal with lists of sbd devices and possibly their respective dumped values, in order to be able to express that.

How does that fit all together?

Reply:
We still haven't decided this, but it could indeed require a bit more thought. This is how it would be specified now (with an alternative if we are not happy with having to provide the SBD device to the gatherer):
```
facts:
  - name: sbd_watchdog_timeout 
    gatherer: sbd_dump
    argument: "/sbd/device:Timeout (watchdog)"
    ## Alternatively, with a bit of work we could have:
    # argument: "Timeout (watchdog)"
    ## to gather the watchdog timeout for all devices (and automatically return an error on the gatherer when they don't match each other)

expectations:
  - name: sbd_watchdog_timeout_correctly_set
    expect: sbd_watchdog_timeout == 60 
  	## or if using the alternative, the same but in plural:
 	# - name: all_sbd_watchdog_timeouts_correctly_set
 	expect: all_sbd_watchdog_timeouts == 60 
```